apps/gdaldem_lib.cpp

```diff
@@ -3642,40 +3749,47 @@ GDALDatasetH GDALDEMProcessing(const char *pszDest, GDALDatasetH hSrcDataset,
         osFormat = psOptions->osFormat;
     }
 
-    GDALDriverH hDriver = GDALGetDriverByName(osFormat);
-    if (hDriver == nullptr ||
-        (GDALGetMetadataItem(hDriver, GDAL_DCAP_CREATE, nullptr) == nullptr &&
-         GDALGetMetadataItem(hDriver, GDAL_DCAP_CREATECOPY, nullptr) ==
-             nullptr))
+    GDALDriverH hDriver = nullptr;
+    if (!EQUAL(osFormat.c_str(), "stream"))
     {
-        CPLError(CE_Failure, CPLE_AppDefined,
-                 "Output driver `%s' not recognised to have output support.",
-                 osFormat.c_str());
-        fprintf(stderr, "The following format drivers are configured\n"
-                        "and support output:\n");
-
-        for (int iDr = 0; iDr < GDALGetDriverCount(); iDr++)
+        hDriver = GDALGetDriverByName(osFormat);
+        if (hDriver == nullptr ||
+            (GDALGetMetadataItem(hDriver, GDAL_DCAP_CREATE, nullptr) ==
+                 nullptr &&
+             GDALGetMetadataItem(hDriver, GDAL_DCAP_CREATECOPY, nullptr) ==
+                 nullptr))
         {
-            hDriver = GDALGetDriver(iDr);
-
-            if (GDALGetMetadataItem(hDriver, GDAL_DCAP_RASTER, nullptr) !=
-                    nullptr &&
-                (GDALGetMetadataItem(hDriver, GDAL_DCAP_CREATE, nullptr) !=
-                     nullptr ||
-                 GDALGetMetadataItem(hDriver, GDAL_DCAP_CREATECOPY, nullptr) !=
-                     nullptr))
+            CPLError(
+                CE_Failure, CPLE_AppDefined,
+                "Output driver `%s' not recognised to have output support.",
```

Comment:
```suggestion
                "Output driver `%s' does not support writing."
```
not sure, but seems clearer to me.

Reply:
done